src/group_theory/exponent.lean

```diff
@@ -255,28 +300,28 @@ begin
   rw list.subperm_ext_iff,
   by_contra' h,
   obtain ⟨p, hp, hpe⟩ := h,
-  haveI hp := fact.mk (nat.prime_of_mem_factors hp),
-  simp only [←padic_val_nat_eq_factors_count p] at hpe,
-  set k := padic_val_nat p (order_of t) with hk,
-  obtain ⟨g, hg⟩ := hp.1.exists_order_of_eq_pow_padic_val_nat_exponent G,
+  replace hp := nat.prime_of_mem_factors hp,
+  simp only [nat.factors_count_eq] at hpe,
+  set k := (order_of t).factorization p with hk,
+  obtain ⟨g, hg⟩ := hp.exists_order_of_eq_pow_factorization_exponent G,
   suffices : order_of t < order_of (t ^ (p ^ k) * g),
   { rw ht at this,
     exact this.not_le (le_cSup hfin.bdd_above $ set.mem_range_self _) },
-  have hpk  : p ^ k ∣ order_of t := pow_padic_val_nat_dvd,
+  have hpk  : p ^ k ∣ order_of t := nat.pow_factorization_dvd _ _,
   have hpk' : order_of (t ^ p ^ k) = order_of t / p ^ k,
-  { rw [order_of_pow' t (pow_ne_zero k hp.1.ne_zero), nat.gcd_eq_right hpk] },
+  { rw [order_of_pow' t (pow_ne_zero k hp.ne_zero), nat.gcd_eq_right hpk] },
   obtain ⟨a, ha⟩ := nat.exists_eq_add_of_lt hpe,
   have hcoprime : (order_of (t ^ p ^ k)).coprime (order_of g),
   { rw [hg, nat.coprime_pow_right_iff (pos_of_gt hpe), nat.coprime_comm],
-    apply or.resolve_right (nat.coprime_or_dvd_of_prime hp.1 _),
+    apply or.resolve_right (nat.coprime_or_dvd_of_prime hp _),
     nth_rewrite 0 ←pow_one p,
-    convert pow_succ_padic_val_nat_not_dvd (h $ t ^ p ^ k),
-    rw [hpk', padic_val_nat.div_pow hpk, hk, nat.sub_self],
-    apply_instance },
+    convert nat.pow_succ_factorization_not_dvd (h $ t ^ p ^ k).ne' hp,
+    rw [hpk', nat.factorization_div hpk],
+    simp [hp] },
```

Comment:
this `simp` working makes me really quite happy; to me, it's meaning that the API is working very well. maybe even `nat.factorization_div` should be a `simp` lemma.

on a sadder note, whilst doing all this cleanup around this file, I ended up adding a bunch of `to_move` lemmas; I'll move them in a bit and maybe start PRs for them separately so that we don't get into a forever-long review process

Reply:
Yeah I'm worried the diff here is getting way too big! great to see these are working though